gosling-screenshot.js

```diff
@@ -91,31 +100,47 @@ async function processFiles(inputDir, outputDir, fileType) {
 }
 
 // Command line argument parsing
-const args = process.argv.slice(2);
+const args = util.parseArgs({
+	allowPositionals: true,
+	options: {
+		format: { type: "string" }, // --format=png
+		outdir: { type: "string" }, // --outdir=output-images
+	},
+});
 
-if (args.length < 2) {
-	console.error(
-		"Usage: node gosling-screenshot.js <input-directory> <output-directory> [file-type]"
-	);
-	console.error("file-type options: png, jpeg, webp (default: png)");
+if (args.positionals.length !== 1) {
+	console.error(`Usage: node gosling-screenshot.js [options] <input>
+
+Options:
+  --format=[png|jpeg|webp] Output image format (default: png)
+  --outdir=<directory>     Directory to save the output image
+                            (required if input is a directory, optional otherwise)
+
+Arguments:
+  input                    A Gosling specification file (JSON) or a directory of specifications`);
 	process.exit(1);
 }
 
-const inputDir = args[0];
-const outputDir = args[1];
-const fileType = args[2]?.toLowerCase() || 'png';
+const input = path.resolve(args.positionals[0]);
+const fileType = args.values.format ?? "png";
+const outputDir = args.values.outdir ?? null;
 
 // Validate file type
-const validFileTypes = ['png', 'jpeg', 'webp'];
+const validFileTypes = ["png", "jpeg", "webp"];
 if (!validFileTypes.includes(fileType)) {
-	console.error(`Invalid file type. Supported types are: ${validFileTypes.join(', ')}`);
+	console.error(
+		`Invalid file type. Supported types are: ${validFileTypes.join(", ")}`,
+	);
 	process.exit(1);
 }
 
-// Process the files
-processFiles(inputDir, outputDir, fileType)
-	.then(() => console.log('Processing complete!'))
-	.catch(err => {
-		console.error('Error:', err);
+const main = fs.statSync(input).isDirectory()
+	? () => processFiles(input, outputDir, fileType)
+	: () => processFile(input, path.dirname(input), fileType);
```

Comment:
Use `outputDir` if provided?

```suggestion
	: () => processFile(input, outputDir  ?? path.dirname(input), fileType);
```